### DISCUSSION TITLE
Computation of PSNR differs from definition on Wikipedia

### DISCUSSION THREAD

**Discussion by bojobo** [photosynthesis-team/piq#367] [Q&A] [answered]
Hi!

The "official" definition given on [Wikipedia](https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio) the calculation of PSNR and the [implementation](https://github.com/photosynthesis-team/piq/blob/c26ce7c78e2a77c1975cb50362befb60c356c6d5/piq/psnr.py#L43) in this package differ a bit.

What is the reason for this? Why is the first part of the subtraction in the last line left out?

**Comment by jzakirov** [comment, marked as answer]
Good catch! Initially we only supported tensors in [0, 1] range and left part of log10(MAX_i) were equal to zero for MAX_i = 1.0. 
We added support for arbitrary ranged tensors later on, but didn't change that part of PSNR implementation. 
~~I'll open a bug report~~

upd: We scale tensors to [0, 1] range in [this line](https://github.com/photosynthesis-team/piq/blob/c26ce7c78e2a77c1975cb50362befb60c356c6d5/piq/psnr.py#L34), so left part in equation is always equal to zero. Implementation is correct.

**Comment by bojobo** [comment]
Good point! I've overlooked that line.
